storage/include/rocksdb/native_client.h

```diff
@@ -0,0 +1,542 @@
+  // Precondition: the iterator points to a key-value pair.
+  std::string key() const;
+  std::string value() const;
+
+  // A view into the key and value inside the iterator. The return value is valid until the iterator is modified.
+  // Precondition: the iterator points to a key-value pair.
+  std::string_view keyView() const;
+  std::string_view valueView() const;
+
+ private:
+  Iterator(std::unique_ptr<::rocksdb::Iterator> iter);
+  static std::string sliceToString(const ::rocksdb::Slice &);
+  static std::string_view sliceToStringView(const ::rocksdb::Slice &);
+
+ private:
+  std::unique_ptr<::rocksdb::Iterator> iter_;
+  friend class NativeClient;
+};
+
+class NativeClient : public std::enable_shared_from_this<NativeClient> {
+ public:
+  static std::shared_ptr<NativeClient> newClient(const std::string &path, bool readOnly);
+
+  // Use this native client through the IDBClient interface.
+  std::shared_ptr<IDBClient> asIDBClient() const;
+
+  // Column family single key read-write interface.
+  template <typename KeySpan, typename ValueSpan>
+  void put(const std::string &cFamily, const KeySpan &key, const ValueSpan &value);
+  // Returns nullopt if the key is not found.
+  template <typename KeySpan>
+  std::optional<std::string> get(const std::string &cFamily, const KeySpan &key) const;
+  // Deleting a key that doesn't exist is not an error.
+  template <typename KeySpan>
+  void del(const std::string &cFamily, const KeySpan &key);
+
+  // Single key read-write interface for the default column family.
+  template <typename KeySpan, typename ValueSpan>
+  void put(const KeySpan &key, const ValueSpan &value);
+  // Returns nullopt if the key is not found.
+  template <typename KeySpan>
+  std::optional<std::string> get(const KeySpan &key) const;
+  // Deleting a key that doesn't exist is not an error.
+  template <typename KeySpan>
+  void del(const KeySpan &key);
+
+  // Batching interface.
+  WriteBatch getBatch() const;
+  void write(WriteBatch &&);
+
+  // Iterator interface.
+  // Iterators initially don't point to a key value, i.e. they convert to false.
+  // Important note - RocksDB requires that iterators are destroyed before the DB client that created them.
+  //
+  // Get an iterator into the default column family.
+  Iterator getIterator() const;
+  // Get an iterator into a column family
+  Iterator getIterator(const std::string &cFamily) const;
+  // Get iterators from a consistent database state across multiple column families. The order of the returned iterators
+  // match the families input.
+  std::vector<Iterator> getIterators(const std::vector<std::string> &cFamilies) const;
+
+  // Column family management.
+  static std::string defaultColumnFamily();
+  static std::unordered_set<std::string> columnFamilies(const std::string &path);
+  std::unordered_set<std::string> columnFamilies() const;
+  // If is not an error if it already exists.
+  void createColumnFamily(const std::string &cFamily);
```

Comment:
What about configuring a column family?

Reply:
Good point. I have added that in a second commit. It required a number of changes to initialisation.